### PR TITLE
Setup encode databases

### PR DESCRIPTION
**Pull request by Pradip-p**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation
- [x] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue
N/A

## Describe this PR

This PR introduces a new  encode database connection implementation and feature for creating projects, upload task boundaries, list all projects, user login & user sign up  using a raw SQL query. It includes the following changes:
- Update an endpoint `/api/projects/create_project` to handle project creation.
- Update an endpoint `/api/projects/{project_id}/upload-task-boundaries` to upload task boundaries
- Update an endpoint `/api/users/login/` to login users
- Update an endpoint `/api/users/signup` to register users
- Utilizes raw SQL queries for inserting and fetching project data.
- Implements a new encode databases connection.

## Screenshots
Screenshot from 2024-06-26 11-48-34
Screenshot from 2024-06-26 11-50-57

## Alternative Approaches Considered

- Considered using an ORM for project creation but opted for raw SQL for performance reasons and finer control over the query execution.

## Review Guide

- **Testing the Change**: To test this change, you can send a POST request to the `/create_project`  & `/{project_id}/upload-task-boundaries` endpoint with the necessary project metadata.
- **Code to Review**: Focus on the endpoint implementation and the database connection/session management logic.

### CODE REVIEW

**Comment by nrjadkry** [comment]
@Pradip-p This Pull request has a merge conflicts. Will you resolve them.

**Review by nrjadkry — APPROVED** [review]

File: src/backend/app/projects/project_crud.py [hunk 1]
```diff
@@ -26,41 +65,37 @@ async def get_project_by_id(
...
-        app_project.outline_geojson = geometry_to_geojson(
-            db_project.outline, {"id": db_project.id}, db_project.id
-        )
-    app_project.tasks = db_project.tasks
-    return app_project
-
-
 async def get_projects(
-    db: Session,
+    db:Database,
```
Comment (spwoodcock) [inline src/backend/app/projects/project_crud.py:69]: Is this type correct?

It seems like it's probably a SessionLocal object, but I could be wrong

**Review by spwoodcock — APPROVED** [review]

File: src/backend/app/projects/project_crud.py [hunk 2]
```diff
@@ -119,26 +154,18 @@ async def create_tasks_from_geojson(
...
-            )
-
-        # Commit all tasks and update project location in db
-        db.commit()
-
-        log.debug("COMPLETE: creating project boundary, based on task boundaries")
-
-        return True
+                ]                  
+            query = f""" INSERT INTO tasks (project_id,outline,project_task_index) VALUES ( '{project_id}', '{wkblib.dumps(shape(polygon["geometry"]), hex=True)}', '{index + 1}');"""
```
Comment (spwoodcock) [inline src/backend/app/projects/project_crud.py:158]: Definitely don't use f-strings to insert values into the db. It probably opens us up to SQL injection attacks.

I'm sure there is probably a recommended way from the Databases docs.

File: src/backend/app/users/user_crud.py [hunk 3]
```diff
@@ -61,47 +61,58 @@ def get_password_hash(password: str) -> str:
...
     return data
 
+async def get_user_email(db: Database, email: str):
+    query = f"SELECT * FROM users WHERE email_address = '{email}' LIMIT 1;"
+    result = await db.fetch_one(query)
+    return result
+
+
+async def get_user_username(db: Database, username: str):
+    query = f"SELECT * FROM users WHERE username = '{username}' LIMIT 1;"
```
Comment (spwoodcock) [inline src/backend/app/users/user_crud.py:77]: Please update all f-string usage for SQL 🙏

**Review by spwoodcock — CHANGES_REQUESTED** [review]

**Pull request merged**